src/Sylius/Bundle/CoreBundle/Announcer/CatalogPromotionAnnouncer.php

```diff
@@ -42,6 +42,10 @@ public function dispatchCatalogPromotionCreatedEvent(CatalogPromotionInterface $
 
     public function dispatchCatalogPromotionUpdatedEvent(CatalogPromotionInterface $catalogPromotion): void
     {
+        if ($catalogPromotion->getStartDate() > $this->dateTimeProvider->now()) {
+            $this->eventBus->dispatch(new CatalogPromotionUpdated($catalogPromotion->getCode()), []);
```

Comment:
```suggestion
            $this->eventBus->dispatch(new CatalogPromotionUpdated($catalogPromotion->getCode()));
```
Due to the fact that the dispatch's method second param has a default value we haven't to pass an empty array

Reply:
I left it intentionally like this to keep it consistent with every other dispatching where we send extra argument even if it should be dispatched right away.